src/OptionBar.js

```diff
@@ -1,9 +1,113 @@
-import React from "react";
+import React, { useCallback, useContext, useState } from "react";
+import IdeaForm from "./IdeaForm";
+import { ThemeContext } from "./ThemeContext";
+import { IdeaContext } from "./IdeaContext";
+import update from "immutability-helper";
+import { v4 as uuidv4 } from "uuid";
+import "./OptionBar.css";
+import { SketchPicker } from "react-color";
 
 const OptionBar = () => {
+  const ideaPackage = useContext(IdeaContext);
+  const selectedId = ideaPackage.selectedId;
+  const selectedIdea = ideaPackage.ideas[selectedId];
+
+  const themePackage = useContext(ThemeContext);
+  const [colorSelectorValue, setColorSelectorValue] = useState(0);
+  const [background, setBackground] = useState("#800080");
+  const [hiding, setHiding] = useState({ display: "none" });
+
+  const changeColor = useCallback(
+    (color) => {
+      setBackground(color.hex);
+      switch (colorSelectorValue) {
+        case "1":
+          themePackage.updateTheme({
+            element: "optionBarColor",
+            color: color.hex,
+          });
+          break;
+        case "2":
+          themePackage.updateTheme({
+            element: "freeFormIdeasColor",
+            color: color.hex,
+          });
+          break;
+        case "3":
+          themePackage.updateTheme({
+            element: "selectedIdeaColor",
+            color: color.hex,
+          });
+          break;
+        default:
+          return;
+      }
```

Comment:
The only thing that changes between all of these cases is the `element` attribute.
```suggestion
      themePackage.updateTheme({
        element: { 1: "optionBarColor", 2: "freeFormIdeasColor", 3: "selectedIdeaColor" }[colorSelectorValue],
        color: color.hex,
      });
```

Reply:
jeez